packages/stream-model-instance-handler/src/__tests__/model-instance-document-handler.test.ts

```diff
@@ -797,6 +857,107 @@ describe('ModelInstanceDocumentHandler', () => {
     expect(state).toMatchSnapshot()
   })
 
+  it('MIDs with SET account relation validate signed commit fields', async () => {
+    const genesisCommit = (await ModelInstanceDocument._makeGenesis(context.api, null, {
+      ...DETERMINISTIC_METADATA,
+      model: FAKE_MODEL_SET_ID,
+      unique: ['one', 'two'],
+    })) as GenesisCommit
+    await context.ipfs.dag.put(genesisCommit, FAKE_CID_1)
+
+    // apply genesis
+    const genesisCommitData = {
+      cid: FAKE_CID_1,
+      type: CommitType.GENESIS,
+      commit: genesisCommit,
+    }
+    let state = await handler.applyCommit(genesisCommitData, context)
+
+    const state$ = TestUtils.runningState(state)
+    const doc = new ModelInstanceDocument(state$, context)
+
+    const signedCommitFail = (await doc._makeCommit(context.api, {
+      one: 'one',
+      two: 'three',
+      myData: 2,
+    })) as SignedCommitContainer
+    await context.ipfs.dag.put(signedCommitFail, FAKE_CID_2)
+
+    const payloadFail = dagCBOR.decode(signedCommitFail.linkedBlock)
+    await context.ipfs.dag.put(payloadFail, signedCommitFail.jws.link)
+
+    const signedCommitDataFail = {
+      cid: FAKE_CID_2,
+      type: CommitType.SIGNED,
+      commit: payloadFail,
+      envelope: signedCommitFail.jws,
+    }
+    await expect(handler.applyCommit(signedCommitDataFail, context, state)).rejects.toThrow(
+      'Unique content fields value does not match metadata. If you are trying to change the value of these fields, this is causing this error: these fields values are not mutable.'
+    )
+
+    const signedCommitOK = (await doc._makeCommit(context.api, {
+      one: 'one',
+      two: 'two',
+      myData: 2,
+    })) as SignedCommitContainer
+    await context.ipfs.dag.put(signedCommitOK, FAKE_CID_3)
+
+    const payloadOK = dagCBOR.decode(signedCommitOK.linkedBlock)
+    await context.ipfs.dag.put(payloadOK, signedCommitOK.jws.link)
+
+    const signedCommitDataOK = {
+      cid: FAKE_CID_3,
+      type: CommitType.SIGNED,
+      commit: payloadOK,
+      envelope: signedCommitOK.jws,
+    }
+    state = await handler.applyCommit(signedCommitDataOK, context, state)
+    expect(state).toMatchSnapshot()
+  })
+
+  it('MIDs with SET account relation validate content schema on update', async () => {
+    const genesisCommit = (await ModelInstanceDocument._makeGenesis(context.api, null, {
+      ...DETERMINISTIC_METADATA,
+      model: FAKE_MODEL_SET_ID,
+      unique: ['a', 'b'],
+    })) as GenesisCommit
+    await context.ipfs.dag.put(genesisCommit, FAKE_CID_1)
+
+    // apply genesis
+    const genesisCommitData = {
+      cid: FAKE_CID_1,
+      type: CommitType.GENESIS,
+      commit: genesisCommit,
+    }
+
+    // Genesis creation should always work independently of content validation
```

Comment:
I don't think it's really a quirk, it's really about how deterministic generation works: we don't really "create" the stream but rather generate the header so we can derive the stream ID to access. When the genesis commit is created/accessed, it can't have content anyway, it's the same behavior with the SINGLE account relation.

Reply:
👍 yes makes sense, I'll update the comment